### ISSUE TITLE
Predictions accuracy ceiling: ML-driven scoring to break 75% barrier

### ISSUE THREAD

**Issue by holden** [razrfly/cinegraph#658]
## Background

After Phases 1–4 of the predictions pipeline (converging to CriteriaScoring), historical validation accuracy is stuck at ~48–49% across all 5 named weight profiles. This is the same ceiling as the old ScoringService approach (49.7%). Hand-tuning weights cannot break through this barrier.

### Bug Fixed: Festival abbreviation mismatch (caused 4.2% regression)
The `score_festival_nomination` function used wrong abbreviations. Fixed in this branch:
- `"CANNES"` → should be `"CFF"` (7,317 nominations in DB)
- `"SUNDANCE"` → should be `"SFF"` (1,116 nominations)
- `"BAFTA"` not mapped (6,670 nominations — world's second-biggest awards)
- `"HFPA"` (Golden Globes) not mapped (5,516 nominations)

This caused 4.2% accuracy (now fixed back to 48.6%).

### Why we're stuck at 48–49%: structural data gaps

Festival data coverage for 1001-list movies by decade:

| Decade | 1001-list movies | With fest noms | Coverage |
|--------|-----------------|----------------|---------|
| 1920s  | 44              | 2              | 4.5%    |
| 1930s  | 89              | 35             | 39.3%   |
| 1940s  | 91              | 51             | 56.0%   |
| 1950s  | 135             | 95             | 70.4%   |
| 1960s  | 156             | 96             | 61.5%   |
| 1970s  | 160             | 103            | 64.4%   |
| 1980s  | 162             | 118            | 72.8%   |
| 1990s  | 154             | 122            | 79.2%   |
| 2000s  | 123             | 113            | 91.9%   |
| 2010s  | 114             | 109            | 95.6%   |

Same problem for `ivory_tower`: Metacritic and RT scores don't exist for most pre-1980 films. With `festival_recognition` at 30% weight and `ivory_tower` at 17.5%, movies lacking this data start 47.5 points behind any film with any festival data. No weight adjustment fixes a missing signal.

### The 5 named profiles all hit the same ceiling

| Profile | Accuracy |
|---------|---------|
| default | 48.6% |
| festival-heavy | 44.5% |
| audience-first | 48.8% |
| critics-choice | 48.5% |
| auteur | ~48% |

The spread is only 4.3 percentage points. Weight tuning is not the lever.

### Strong signals that exist for ALL movies in the DB

1. **Other canonical list membership** (`canonical_sources` JSONB) — A movie on Sight & Sound 2022 *and* TSPDT 1000 *and* Criterion is overwhelmingly likely to join the 1001 list. We have 9 lists in the DB.
2. **IMDb vote count** — A proxy for cultural staying power. Available for nearly all movies. High vote count (>100K) on a film from 1965 = enduring cultural relevance.
3. **IMDb rating** (already in `mob`) — Solid, universal signal
4. **Director's 1001-list filmography** (already in `auteur_recognition`) — Reliable for established directors

### Phase 5A: Lens Validation — Do the 6 criteria actually work?

Before optimizing weights, verify each scoring lens produces accurate, calibrated scores on movies where we KNOW the ground truth.

**Approach:** For each criterion, sample movies that should clearly score high and low, and verify the scores are correct. This is a data quality audit.

#### Lens 1: The Mob (audience ratings)
- **Verify:** Do movies with high IMDb ratings (8.5+) score 80-90+ on mob?
- **Check:** Are IMDb ratings being correctly converted from 0-10 to 0-100?
- **Data gaps:** How many movies lack IMDb data entirely? TMDb/RT data?
- **Known issue:** The 0-10 → 0-100 conversion exists but is it accurate?

#### Lens 2: Ivory Tower (critic ratings)
- **Verify:** Do Metacritic 90+ movies score 90+ on ivory_tower?
- **Data gaps:** Most pre-1980 films lack Metacritic/RT — how many in total?
- **Risk:** This lens is nearly empty for classic films and cannot be filled without API work

#### Lens 3: Festival Recognition
- **Verify:** Does "Parasite" (Palme d'Or 2019) score 95+? Does "Roma" (Venice 2018) score 90+?
- **Fixed bug:** Festival abbreviations now correctly map CFF, BAFTA, HFPA, SFF
- **Remaining risk:** Are prestigious Cannes/Venice/Berlin categories (director prize, special jury) correctly categorized?
- **Data gaps:** 1920s-1940s coverage is 4.5%-56%, structurally unreliable

#### Lens 4: Cultural Impact
- **Verify:** Does "Titanic" (huge ROI, massive IMDb votes) score high? Does an obscure arthouse film score low?
- **Sub-scores:** ROI (budget/revenue), popularity (IMDb votes), genre diversity, international impact
- **Known issue:** Budget/revenue data in TMDb is often 0 or wrong for older films

#### Lens 5: Technical Innovation
- **Verify:** Does "2001: A Space Odyssey" score high (multiple technical Oscars)?
- **Risk:** Category name matching (`LIKE '%cinematography%'`) — are our festival category names in this format?
- **Check:** Run a query to see which movies currently score >0 on technical innovation

#### Lens 6: Auteur Recognition
- **Verify:** Does a Bergman film score 100 (he has 10+ on the 1001 list)? Does a first-time director score 20?
- **Risk:** Are director credits reliably linked in movie_credits table?
- **This lens is the most reliable** because it uses fully structured data we control

**Deliverable:** SQL query report showing per-criterion score distributions, % of movies scoring 0, and spot-check of known films.

### Phase 5B: ML-Based Feature Discovery

Use the Mac Studio M4 (64GB RAM) to run proper ML analysis in Python instead of 20-minute Elixir backtests.

**Why this is necessary:** The 5 current profiles span only 4.3% accuracy (44.5% to 48.8%). Manual weight tuning in this space is effectively random. ML with SHAP will tell us which signals actually separate 1001-list films from non-list films, and which signals are noise.

**Toolchain (mirrors volfefe project):**
```
polars + connectorx  → load entire DB to Parquet in ~60s
xgboost              → find optimal feature combination
shap                 → understand which features matter
mlflow               → track all experiments
scikit-learn         → cross-validation per decade
```

**Feature matrix to export:**
```
movie_id, decade, is_on_1001_list (label)
mob_score, ivory_tower_score, festival_recognition_score
cultural_impact_score, technical_innovation_score, auteur_recognition_score
canonical_overlap_count    ← NEW: how many other canonical lists
imdb_votes_log             ← NEW: cultural staying power
has_festival_data          ← diagnostic: does missing data hurt?
has_ivory_tower_data       ← diagnostic
```

**Expected insight:** SHAP analysis will show if `canonical_overlap_count` or `imdb_votes_log` are stronger predictors than the current 6 criteria. If so, adding them as features will break through the 48% ceiling.

**Iteration speed:** Each weight experiment: 5-30 seconds (vs 20 minutes for full Elixir backtest). Enables hundreds of experiments per day.

### Phase 5C: Optimal Weight Discovery & Calibration

Once Phase 5B identifies the right signals, use the ML-discovered weights to:

1. **Update named profiles** in `criteria_scoring.ex` with data-driven weights
2. **Add canonical_overlap_score** as a 7th scoring criterion if SHAP shows it's important
3. **Decade-aware weight profiles** if signals differ by era (festival is weak pre-1960, canonical_overlap is strong for all eras)
4. **Report target accuracy** per decade and overall

**Draft: canonical overlap signal (pending validation in Phase 5A)**
```elixir
def score_canonical_overlap(movie) do
  sources = movie.canonical_sources || %{}
  other_lists = Map.delete(sources, "1001_movies")
  # Each canonical list = 15 pts, max 100
  min(map_size(other_lists) * 15.0, 100.0)
end
```

## Hardware & Tooling Notes

- **Mac Studio M4, 64GB RAM**: XGBoost on 100K movies is trivially fast on CPU. No GPU needed.
- **connectorx**: Direct PostgreSQL → Polars/Arrow. Much faster than SQLAlchemy + pandas.
- **Parquet caching**: Export once, iterate on weights in seconds.
- **Parallel processing**: Current Elixir backtest is sequential per decade. Python can parallelize trivially.
- **MLflow UI**: `mlflow ui` shows all experiments, allows comparison — no more guessing which weight combination was best.

## Files to Create (Python side, dev-only)

```
scripts/
  export_prediction_features.py   # PostgreSQL → Parquet export
  find_optimal_weights.py         # XGBoost weight discovery + LOO CV
  shap_analysis.py                # Feature importance visualization
  requirements.txt                # polars, connectorx, xgboost, shap, mlflow
```

No changes to production Elixir code until Phase 5C confirms what to change.

## Success Metrics

- Phase 5A: Each criterion verified with spot-checks; data gap report produced
- Phase 5B: SHAP analysis identifies top-3 features; iteration time < 60s
- Phase 5C: Historical validation accuracy > 75% on 1960s–2010s decades

## Notes

- The 4.2% regression and its fix (festival abbreviations) are already committed
- 75% may be achievable with canonical_overlap signal alone — hypothesis to test in Phase 5B
- Historical accuracy for 1920s-1940s will likely remain lower due to genuine data sparsity; this is acceptable

**Comment by holden** [comment]
SUPERSEDED by #682 — same theme, #682 is the active version.
